### PR TITLE
8345057: ML_KEM NamedParameterSpec constants removed by ML-DSA integration

### PR DESCRIPTION
**Pull request by valeriepeng**
Could someone please help review this trivial fix?

The constants for ML-KEM are accidentally removed during merge. This PR adds them back and adds a test for checking them against the expected ones.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345057](https://bugs.openjdk.org/browse/JDK-8345057): ML_KEM NamedParameterSpec constants removed by ML-DSA integration (**Bug** - P2)

### Reviewers
 * [Sean Mullan](https://openjdk.org/census#mullan) (@seanjmullan - **Reviewer**)
 * [Hai-May Chao](https://openjdk.org/census#hchao) (@haimaychao - Committer) Review applies to [4e0640a0](https://git.openjdk.org/jdk/pull/22397/files/4e0640a0f8cd850611e880ca319c53d718fca79b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22397/head:pull/22397` \
`$ git checkout pull/22397`

Update a local copy of the PR: \
`$ git checkout pull/22397` \
`$ git pull https://git.openjdk.org/jdk.git pull/22397/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22397`

View PR using the GUI difftool: \
`$ git pr show -t 22397`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22397.diff">https://git.openjdk.org/jdk/pull/22397.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22397#issuecomment-2501723643)
</details>

### CODE REVIEW

File: test/jdk/java/security/spec/TestNamedParameterSpec.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            System.out.println("expected: " + Arrays.toString(EXPECTED));
+            System.out.println("actual: " + Arrays.toString(actual));
+            throw new RuntimeException("Name check failed");
+        }
+        System.out.println("Test Passed");
+    }
+
+    public static String[] getSortedConstNames(Class<?> clazz) {
+        TreeSet<String> names = new TreeSet<String>();
+        for(Field field : clazz.getDeclaredFields()){
```
Comment (seanjmullan) [inline test/jdk/java/security/spec/TestNamedParameterSpec.java:60]: Nit: add space after "for".

File: test/jdk/java/security/spec/TestNamedParameterSpec.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+/*
+ * @test
+ * @bug 8345057
+ * @summary Test the existence of constants inside the
+ *         java.security.spec.NamedParameterSpec class.
+ * @run main TestNamedParameterSpec
+ */
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.security.spec.*;
```
Comment (seanjmullan) [inline test/jdk/java/security/spec/TestNamedParameterSpec.java:33]: Nit: can you just import `NamedParameterSpec`?

**Comment by valeriepeng** [comment]
> A few small comments but this looks fine in order to get this fixed quickly. However, I think it would be better if there were ML-KEM and ML-DSA specific tests that used the public constants. I think we should followup on that with Ben and Rajan and possibly open a separate issue to improve existing tests.

Yes, that makes sense. This is just for sanity checking this fix.

**Review by haimaychao — APPROVED** [review]

**Review by seanjmullan — APPROVED** [review]

**Pull request closed without merging**